internal/legacy/v040/migrate.go

```diff
@@ -0,0 +1,297 @@
+		// unmarshal relative source genesis application state
+		var govGenState v036gov.GenesisState
+		v039Codec.MustUnmarshalJSON(appState[v036gov.ModuleName], &govGenState)
+
+		// delete deprecated x/gov genesis state
+		delete(appState, v036gov.ModuleName)
+
+		// Migrate relative source genesis application state and marshal it into
+		// the respective key.
+		appState[v040gov.ModuleName] = v040Codec.MustMarshalJSON(v040gov.Migrate(govGenState))
+	}
+
+	// Migrate x/mint.
+	if appState[v039mint.ModuleName] != nil {
+		// unmarshal relative source genesis application state
+		var mintGenState v039mint.GenesisState
+		v039Codec.MustUnmarshalJSON(appState[v039mint.ModuleName], &mintGenState)
+
+		// delete deprecated x/mint genesis state
+		delete(appState, v039mint.ModuleName)
+
+		// Migrate relative source genesis application state and marshal it into
+		// the respective key.
+		appState[v040mint.ModuleName] = v040Codec.MustMarshalJSON(v040mint.Migrate(mintGenState))
+	}
+
+	// Migrate x/slashing.
+	if appState[v039slashing.ModuleName] != nil {
+		// unmarshal relative source genesis application state
+		var slashingGenState v039slashing.GenesisState
+		v039Codec.MustUnmarshalJSON(appState[v039slashing.ModuleName], &slashingGenState)
+
+		// delete deprecated x/slashing genesis state
+		delete(appState, v039slashing.ModuleName)
+
+		// Migrate relative source genesis application state and marshal it into
+		// the respective key.
+		appState[v040slashing.ModuleName] = v040Codec.MustMarshalJSON(v040slashing.Migrate(slashingGenState))
+	}
+
+	// Migrate x/staking.
+	if appState[v038staking.ModuleName] != nil {
+		// unmarshal relative source genesis application state
+		var stakingGenState v038staking.GenesisState
+		v039Codec.MustUnmarshalJSON(appState[v038staking.ModuleName], &stakingGenState)
+
+		// delete deprecated x/staking genesis state
+		delete(appState, v038staking.ModuleName)
+
+		// Migrate relative source genesis application state and marshal it into
+		// the respective key.
+		stakingGenesis := v040staking.Migrate(stakingGenState)
+
+		stakingGenesis.Params.HistoricalEntries = 10000
+		appState[v040staking.ModuleName] = v040Codec.MustMarshalJSON(stakingGenesis)
+	}
+
+	// Migrate x/genutil
+	if appState[v039genutil.ModuleName] != nil {
+		// unmarshal relative source genesis application state
+		var genutilGenState v039genutil.GenesisState
+		v039Codec.MustUnmarshalJSON(appState[v039genutil.ModuleName], &genutilGenState)
+
+		// delete deprecated x/staking genesis state
+		delete(appState, v039genutil.ModuleName)
+
+		// Migrate relative source genesis application state and marshal it into
+		// the respective key.
+		appState[v040gentypes.ModuleName] = v040Codec.MustMarshalJSON(migrateGenutil(genutilGenState))
+	}
+
+	// Default capabilities initialization
+	capGenesis := captypes.DefaultGenesis()
+	appState[captypes.ModuleName] = v040Codec.MustMarshalJSON(capGenesis)
+
+	/*
+		TODO: default configuration for IBC
```

Comment:
this is a gaia config... it is not clear that we actually want this.  I filed an issue #102 for future follow up.